fe/fe-core/src/main/java/com/starrocks/planner/PlanFragment.java

```diff
@@ -190,6 +202,29 @@ public void setParallelExecNum(int parallelExecNum) {
         this.parallelExecNum = parallelExecNum;
     }
 
+    public void setPipelineDopIfPipelineEngineEnabled() {
+        if (ConnectContext.get() == null || !ConnectContext.get().getSessionVariable().isEnablePipelineEngine()) {
+            return;
+        }
+        int dop = ConnectContext.get().getSessionVariable().getPipelineDop();
+        this.pipelineDop = dop > 0 ? dop : 1;
```

Comment:
If dop is 0, the pipelineDop should be half num_hardware_cores?

Reply:
For a PlanFragment, in default, parallelExecNum is set to DegreeOfParallelism(half the average number of hardware cores of backends) instead of parallel_fragment_exec_instance_num  and pipelineDop is set to 1.
later these two fields are changed for some PlanFragments. here just the default setting.